Update core.selectbox.cshtml

jonrandahl opened the pull request:
size
If the control is presented as a scrolled list box, this attribute represents the number of rows in the list that should be visible at one time. Browsers are not required to present a select element as a scrolled list box. The default value is 0.

https://developer.mozilla.org/en/docs/Web/HTML/Element/select

kjac commented:
Hey @jonrandahl,

Thanks :) I'll merge it in later today.

kjac commented:
@jonrandahl OK so while reviewing this, I realized that I actually had a reason for explicitly setting ```size="@Model.FieldValues.Count()"```. The "Select box" field represents a ```<select>``` that displays all available options (with optional multi select capability), while the "Drop-down" field represents a ```<select>``` as a drop-down menu. 

One could consider if the "Select box" field should have a setting for "number of visible options". Thoughts?

jonrandahl commented:
Kenn,
I think that's a much better option than mine. Having said that, I didn't realise there was a "dropdown" field control as I'm always looking for the select element. It may be a good idea to change the name to "Multiple Select Box" as "Select Box" does make me think of "Dropdown"?

Also, do you believe that it would be a good idea - in the asynchronous form - to either not have a response page but a response container instead OR to be able to load the response page as a partial not a full refresh of the page? In the instance I'm creating, I've added a new tab to the doctype and added the title and content fields there for my "thank you" response and copied and edited the FormEditoAsync.js file to hide the form on submit and show the response container which holds the Umbraco fields?

kjac commented:
@jonrandahl makes sense - both things :)

I'll rename "Select box" to "Multiple options" to avoid confusing others.

A response container could be a nice touch, but it'd be rudimentary to say the least - like a text field for a header and a textarea for some text. I can't very well add a full blown RTE because it'd potentially break the page design for sites, where the RTE's on other content types have been locked down to basic editing capability only.

jonrandahl commented:
Kenn,
With the naming, I think the addition of the "Multiple" keyword to the standard "Select box" would suffice only as I think, from a layman's terms, "options" means something entirely different?

With the async response, I completely agree about the RTE although I can see options for it? Would you like me to send you the updated version of FormEditorAsync.js file I've used? I've hooked an "ng-switch" onto the form and the response container to hide and show respectively. You could probably thing of a better way but this has worked for me.

kjac commented:
@jonrandahl I'm gonna rename "Select box" to "Select box (multiple)" and "Drop-down" to "Select box (drop-down)". Best way around this naming issue, I think.

I'll add an option for a simple receipt message one of these days. 

I'm going to go ahead and close these PRs and create new feature requests for the renaming and the receipt tasks (for tracking).

Oh and thanks for offering, but I think I have a pretty good idea on how to handle the templating part of this :)

kjac commented:
See #12 and #13

jonrandahl commented:
@kjac ,
As for the offer, I certainly agree that you've got a better grasp of it all! ;-)
Thanks for letting me help in this small way, and also for a fantastic package! - I look forward to the update!
J

kjac commented:
@jonrandahl Well no matter what it was nice of you to offer. And thanks a lot for the feedback, it is most appreciated.
